src/main/java/jenkins/metrics/api/MetricProvider.java

```diff
@@ -51,6 +51,16 @@ protected static MetricSet metrics(Map.Entry<String, Metric>... metrics) {
         return new FixedMetricSet(result);
     }
 
+    protected static MetricSet metrics(Map<String, Metric> metrics) {
+        final Map<String, Metric> result = new LinkedHashMap<String, Metric>(metrics.size());
```

Comment:
looks to me like this could almost just call ```return metrics(metrics.entrySet())``` instead of duplicating the above implementation.